lib/parsec/exceptions.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2019 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from copy import copy
+import os
+
+from parsec.util import itemstr
+
+
+class ParsecError(Exception):
+    """Generic exception for Parsec errors."""
+
+    def __str__(self):
+        return '%s: %s' % (self.__class__.__name__, self.args[0])
+
+
+class ItemNotFoundError(ParsecError, KeyError):
```

Comment:
Is this a sensible way to provide more context to the exception or a potential hazard?

Reply:
I think so, knowing it's due/related to `KeyError` looks simple and useful.